packages/services/src/view-service/fees.ts

```diff
@@ -4,7 +4,22 @@ import { assetIdFromBaseDenom } from '@penumbra-zone/wasm/asset';
 import { IndexedDbInterface } from '@penumbra-zone/types/indexed-db';
 
 // Attempts to extract a fee token, with priority in descending order, from the assets used
-// in the actions of the planner request.
+// in the actions of the transaction planner request (TPR).
+//
+// (1) For outputs, spends, swaps, scheduling auctions, and ICS withdrawals, the naive approach
+// of "using the first alternative token for which the user has a balance" is semantically
+// equivalent to "extract the input asset from the transaction request".
```

Comment:
Are these actually the equivalent? They sound different. One sounds like it takes from balances and the other takes from the input, lessening the effect of say a withdraw.

Reply:
realize that comment needs some clarity:

"For outputs, spends, swaps, scheduling auctions, and ICS withdrawals, the naive approach of "using the first alternative token for which the user has a balance" is similar to "extracting the input asset from the transaction request". While both methods aim to retrieve an alternative fee token, they may result in selecting different tokens."

Requesting an alternative token can be queried from the incoming request or the balance. The former provides more consistency, taking the input asset as the alternative fee (eg. transfer of GM will pay for fees in GM). The latter may result in some intuitive fee selection (eg. transfer of GM might pay for fees in GN).

Reply:
simplified entire description:
```
// Attempts to extract a fee token, with priority in descending order, from the assets used
// in the actions of the transaction planner request (TPR). If no fee token is found from the
// specified assets, it falls back to checking the gas prices table for an asset with a positive balance.
```